scrapi/base/transformer.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import unicode_literals
+
+import abc
+import logging
+from copy import deepcopy
+from functools import partial
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTransformer(object):
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, schema):
+        self.schema = deepcopy(schema)
+
+    def transform(self, doc):
+        self.process_schema()
+        return self._transform(self.schema, doc)
+
+    def _transform(self, schema, doc):
+        ret = {}
+        for key, transformation in schema.items():
+            if isinstance(transformation, dict):
+                ret[key] = self._transform(transformation, doc)
+            else:
+                ret[key] = transformation(doc)
+
+        return ret
+
+    @abc.abstractmethod
+    def process_schema(self):
+        raise NotImplementedError
+
+
+class XMLTransformer(BaseTransformer):
+
+    def __init__(self, name, schema, namespaces):
+        super(XMLTransformer, self).__init__(schema)
+        self.namespaces = namespaces
+        self.NAME = name
+
+        self._processed = False
+
+    def process_schema(self):
+        if not self._processed:
```

Comment:
is `self._processed` first defined here? Maybe make it `None` in either the class or in `__init__`